eng/Configurations.props

```diff
@@ -52,7 +52,8 @@
     <TargetOS Condition="'$(TargetOS)' == '' and $([MSBuild]::IsOSPlatform('OSX'))">OSX</TargetOS>
     <TargetOS Condition="'$(TargetOS)' == '' and $([MSBuild]::IsOSPlatform('FREEBSD'))">FreeBSD</TargetOS>
     <TargetOS Condition="'$(TargetOS)' == '' and $([MSBuild]::IsOSPlatform('NETBSD'))">NetBSD</TargetOS>
-    <TargetOS Condition="'$(TargetOS)' == '' and $([MSBuild]::IsOSPlatform('SUNOS'))">SunOS</TargetOS>
+    <TargetOS Condition="'$(TargetOS)' == '' and $([MSBuild]::IsOSPlatform('ILLUMOS'))">Illumos</TargetOS>
```

Comment:
What was wrong with using SUNOS as the name for all the flavors?

Reply:
Illumos and Oracle Solaris are two forks of OpenSolaris which were created in 2010.

Despite of many similaries, binaries compiled on an Illumos distro are not compatible with Oracle Solaris and vice versa. They differ by:

* `libc` and its version; to begin with, dlopen/dlsym specially look for `ILLUMOS xyz` when we try to execute Illumos binary on Solaris.
* supported syscalls; made by the libc itself (before the user code even begins).

Like Darwin is to macOS, iOS, tvOS etc. some utilities and tools such as `uname`, `python` and `cmake` still use the legacy SunOS name on both platforms for compatibility reasons. However, on Illumos `uname -o` returns `illumos` while Oracle Solaris does not support `-o` option.

Reply:
> binaries compiled on an Illumos distro are not compatible with Oracle Solaris and vice versa

The same is true e.g. between Ubuntu and Alpine. We do not have different TargetOS for these two.

Reply:
True. On the other hand, their history is also similar to how OS X and FreeBSD used to share same base, but now they are diverged. Although there are some similarities in API usage, we consider them two different platforms on TargetOS level.

Reply:
If I am not mistaken modern binaries from Solaris 11+ don't run at all on illumos and the reverse is also true. Some older binaries from the Solaris 10 days might still work on both though.

Treating both the same has caused a lot of ugly hacks in Makefiles over the years, every few years there is also talk from dropping SunOS entirely from uname. But as of now that would break too many software that treat both the same.

For new languages (.e.g rust, go, ...) we are treating illumos seperate. Although IIRC for go illumos is a child of sunos (but I can't quickly find the page I was thinking of to confirm this)